### PR TITLE
Prevent passing slot names as props

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Ignore the `slot` prop in Vue components
- Fixes https://github.com/withastro/astro/issues/8921

## Testing

- Test case added

## Docs

N/A, bug fix

### CODE REVIEW

**Review by natemoo-re — APPROVED** [review]
Works for me!

File: .changeset/swift-suits-drum.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Prevent passing slot names as props
```
Comment (sarah11918) [inline .changeset/swift-suits-drum.md:5]:
```suggestion
Fixes an issue where Astro slot names were being rendered as attributes in components. Astro slot names will no longer be sent as props to framework components.
```
 Just tried to tie it into the issue reported. Something like this?

**Review by sarah11918 — COMMENTED** [review]
Just a suggestion to tie the changeset description back to the behaviour reported in the original issue!

**Comment by matthewp** [comment]
This actually might be wrong, putting back into draft.

**Comment by matthewp** [comment]
I had to go with a Vue-specific solution because we don't know what slots are going to be used inside of the component providing the slots.

**Pull request merged**